### DISCUSSION TITLE
Wifi connection issue after fresh installation

### DISCUSSION THREAD

**Discussion by rona1707** [MiczFlor/RPi-Jukebox-RFID#2245] [Q&A] [answered]
Hello together,

since I've trouble with my Wifi connection on my old setup (the box is not connection to my local wifi). I've tried a setup from scratch. The OS (bullseye on RPi 3B+) is installing fine and I'm able to connect to my Wifi without issues. I was running the installation commandline and used the interactive configuration. I tried following combinations:

1. Wifi with static IP and Auto-Hot-Spot **enabled**
2. Wifi with static IP and Auto-Hot-Spot **disabled**
3. no Wifi config (as it is already working before installation) and Auto-Hot-Spot **enabled**
4. no Wifi config (as it is already working before installation) and Auto-Hot-Spot **disabled**

In all cases I end up that the Pi is not connecting to my Wifi **after** the phoniebox installation. Just the Auto-Hot-Spot is kicking in (if configured). The wpa_supplicant.conf seems fine. I've tried to set the Wifi again in the raspi-config without luck. I also tried changing between dhcpcd and NetworkManager and edited the dhcpcd.conf for static ip. Nothing worked out at the end.

Has anyone else faced similar issues? I'd be very happy if someone could help me out on this. I just like to know a way to get the phoniebox to my local wifi and use the hotspot anywhere else.

Regards, Robert

**Comment by AlvinSchiller** [comment]
Hi,
i never experience such problems. 
Which version do you use exactly? And could you provide the outcome of `tail -n 200 /var/log/syslog` and maybe your installation log?

**Comment by rona1707** [comment, marked as answer]
Hi,

I messed around with the installation of yesterday so I tried a fresh one again. The Raspberry Pi OS is "bullseye" Version: 2023-12-05. It seems that after the phoniebox installation the Pi has some issues with my **hidden** Wifi. The Wifi connection with the hidden wifi worked before the installation. The syslog gave me the hint on the issue
Thx @AlvinSchiller for the hint with the syslog :-)

```
Feb  6 20:40:21 phoniebox systemd[1]: Started Access point and authentication server for Wi-Fi and Ethernet.
Feb  6 20:40:21 phoniebox systemd[1]: Started Samba SMB Daemon.
Feb  6 20:40:21 phoniebox systemd[1]: Reached target Multi-User System.
Feb  6 20:40:21 phoniebox systemd[1]: Reached target Graphical Interface.
Feb  6 20:40:21 phoniebox systemd[1]: Starting Automatically generates an internet Hotspot when a valid ssid is not in range...
Feb  6 20:40:21 phoniebox systemd[1]: Starting Update UTMP about System Runlevel Changes...
Feb  6 20:40:21 phoniebox systemd[1]: systemd-update-utmp-runlevel.service: Succeeded.
Feb  6 20:40:21 phoniebox systemd[1]: Finished Update UTMP about System Runlevel Changes.
Feb  6 20:40:21 phoniebox autohotspot[1755]: Device availability check: try 0
Feb  6 20:40:21 phoniebox autohotspot[1755]: wifi device available
Feb  6 20:40:22 phoniebox autohotspot[1755]: SSID availability check: try 0
Feb  6 20:40:22 phoniebox autohotspot[1755]: SSID's in range:
...
Feb  6 20:40:22 phoniebox autohotspot[1755]: BSS <MY_ROUTER_MAC>(on wlan0)
Feb  6 20:40:22 phoniebox autohotspot[1755]: #011SSID:
...
Feb  6 20:40:22 phoniebox packagekitd[1663]: http://raspbian.raspberrypi.org/raspbian bullseye InRelease is not (yet) available (Temporary failure resolving 'raspbian.raspberrypi.org')
Feb  6 20:40:22 phoniebox packagekitd[1663]: http://archive.raspberrypi.org/debian bullseye InRelease is not (yet) available (Temporary failure resolving 'archive.raspberrypi.org')
Feb  6 20:40:22 phoniebox PackageKit: refresh-cache transaction /7_eedbdacd from uid 1000 finished with failed after 3640ms
Feb  6 20:40:22 phoniebox autohotspot[1755]: Hostspot already active
Feb  6 20:40:22 phoniebox systemd[1]: Finished Automatically generates an internet Hotspot when a valid ssid is not in range.
Feb  6 20:40:22 phoniebox systemd[1]: Startup finished in 3.869s (kernel) + 1min 41.340s (userspace) = 1min 45.209s.
Feb  6 20:40:23 phoniebox dhcpcd[547]: eth0: no IPv6 Routers available
```

The following log now shows that the wifi connection could be established successfully. Another trap was the reboot of my router. The configuration interface gave me the option to enable/disable SSID hide on-the-fly. Didn't worked out until the router restart.

```
Feb  6 20:48:34 phoniebox systemd[1]: Starting Automatically generates an internet Hotspot when a valid ssid is not in range...
Feb  6 20:48:34 phoniebox systemd[985]: Starting Virtual filesystem service...
Feb  6 20:48:34 phoniebox systemd[1]: Starting Update UTMP about System Runlevel Changes...
Feb  6 20:48:34 phoniebox systemd[1]: systemd-update-utmp-runlevel.service: Succeeded.
Feb  6 20:48:34 phoniebox systemd[1]: Finished Update UTMP about System Runlevel Changes.
Feb  6 20:48:35 phoniebox dbus-daemon[1088]: [session uid=1000 pid=1088] Successfully activated service 'org.gtk.vfs.Daemon'
Feb  6 20:48:35 phoniebox systemd[985]: Started Virtual filesystem service.
Feb  6 20:48:35 phoniebox autohotspot[1226]: Device availability check: try 0
Feb  6 20:48:35 phoniebox autohotspot[1226]: wifi device available
Feb  6 20:48:35 phoniebox autohotspot[1226]: SSID availability check: try 0
Feb  6 20:48:35 phoniebox autohotspot[1226]: SSID's in range:
...
Feb  6 20:48:35 phoniebox autohotspot[1226]: BSS <MY_ROUTER_MAC>(on wlan0)
Feb  6 20:48:35 phoniebox autohotspot[1226]: #011SSID: <MY_WIFI_SSID>
...
Feb  6 20:48:35 phoniebox autohotspot[1226]: Valid SSID Detected, assesing Wifi status
Feb  6 20:48:36 phoniebox autohotspot[1226]: Connecting to the WiFi Network
Feb  6 20:48:36 phoniebox autohotspot[1226]: Checking WiFi connection ok
Feb  6 20:48:37 phoniebox dbus-daemon[1088]: [session uid=1000 pid=1088] Activating via systemd: service name='org.gtk.vfs.UDisks2VolumeMonitor' unit='gvfs-udisks2-volume-monitor.service' requested by ':1.11' (uid=1000 pid=1273 comm="pcmanfm --desktop --profile LXDE-pi ")
Feb  6 20:48:37 phoniebox systemd[985]: Starting Virtual filesystem service - disk device monitor...
Feb  6 20:48:37 phoniebox systemd[1]: systemd-fsckd.service: Succeeded.
Feb  6 20:48:37 phoniebox dhcpcd[552]: wlan0: carrier acquired
Feb  6 20:48:37 phoniebox dhcpcd[552]: wlan0: connected to Access Point `<MY_WIFI_SSID>'
Feb  6 20:48:37 phoniebox kernel: [   40.852435] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
Feb  6 20:48:37 phoniebox dbus-daemon[1088]: [session uid=1000 pid=1088] Successfully activated service 'org.gtk.vfs.UDisks2VolumeMonitor'
Feb  6 20:48:37 phoniebox systemd[985]: Started Virtual filesystem service - disk device monitor.
Feb  6 20:48:37 phoniebox dhcpcd[552]: wlan0: IAID eb:7f:9e:bf
Feb  6 20:48:37 phoniebox dhcpcd[552]: wlan0: adding address fe80::8e0c:405:5fa3:278
```

**Reply by AlvinSchiller** [reply]
Glad it works 👍
And this is a good insight. I wasn't aware of that.

I just checked the code again and there seems a way to address hidden ssids if you make changes in the script:

https://github.com/AlvinSchiller/RPi-Jukebox-RFID/blob/4101984b1031996239903796ae8fc4ce180106a5/misc/sampleconfigs/autohotspot.sh.stretch-default2-Hotspot.sample#L43-L45

**Reply by rona1707** [reply]
Whats a little bit wierd about it is that I also seem to have connection issue without the Hotspot. Maybe I need to test it again.

**Reply by AlvinSchiller** [reply]
Is your WiFi 2.4 or 5 GHz?
Can you try and set the WiFi location via the raspi-config if you have 5 GHz? I'm not sure if this is setup correctly in all cases.

**Reply by rona1707** [reply]
I've connected the RPi to the 2.4 Ghz during the burning of th SD-Card by the Flashtool. Works for now 👍. Since the rest of the sync from my old installation (GPIO and audiofolders with RFID)  is working now, I'd like to stay at the current installation. The system is running ;-)

Thanks for the help
